File: mfg-shop-react/src/Service/productService.js
import axios from "axios";

export const getAllProduct = async (value) => {
  try {
    const result = await axios.get(
      "http://localhost:8080/api/public/product/list?search=" +
        value.search +
        "&page=" +
        value.page +
        "&typeProduct=" +
        value.typeProduct
    );
    return result.data;
  } catch (e) {
    console.log(e);
  }
};
export const getProductById = async (id) => {
  try {
    const result = await axios.get(
      "http://localhost:8080/api/public/product/" + id
    );
    return result.data;
  } catch (e) {
    console.log(e);
  }
};
export const getProductByTypeProduct = async (id) => {
  try {
    const result = await axios.get(
      "http://localhost:8080/api/public/product/type-product/" + id
    );
    return result.data;
  } catch (e) {
    console.log(e);
  }
};

File: mfg-shop-api/src/main/java/com/mfgshopapi/service/impl/ProductService.java
package com.mfgshopapi.service.impl;

import com.mfgshopapi.model.Product;
import com.mfgshopapi.repository.IProductRepository;
import com.mfgshopapi.service.IProductService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ProductService implements IProductService {
    @Autowired
    private IProductRepository iProductRepository;
    @Override
    public Page<Product> findAll(String search, Pageable pageable) {
        return iProductRepository.findProductsByNameProductContaining(search, pageable);
    }

    @Override
    public Page<Product> findAllByTypeProduct(String search, int typeProduct, Pageable pageable) {
        return iProductRepository.findProductsByNameProductContainingAndTypeProductId(search, typeProduct, pageable);
    }

    @Override
    public Product findById(int id) {
        return iProductRepository.findById(id).get();
    }

    @Override
    public Page<Product> findProductsByTypeProduct(int id, Pageable pageable) {
        return iProductRepository.findProductsByTypeProductId(id, pageable);
    }
}
